### PR TITLE
docs: Add docs for docker usage

### PR DESCRIPTION
**Pull request by karmab**
the corresponding docker image lives at docker.io/karmab/mgmt:0.0.14

### CODE REVIEW

**Comment by purpleidea** [comment]
@karmab Thanks! Karim!

CI failed because we require a prefix for commits: eg: `docs: Add docs for docker usage` or something.
Could you instead add these docs to docs/docker-guide.md instead please?

Cheers!

**Comment by purpleidea** [comment]
@karmab Sorry I did not see your update sooner. GH doesn't notify new pushes...

In any case, build failed because of an extra newline at the end of the dockerfile:

https://travis-ci.org/purpleidea/mgmt/jobs/334697863#L787

Do you mind fixing and updating please? Also can you move the Dockerfile into the docker/ folder?

Thank you! (Ping me here when it's done so I can merge)

File: Dockerfile.static [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM centos:7
```
Comment (purpleidea) [inline Dockerfile.static]: Can you move the docker files into docker/ folder please?

File: docs/quick-start-guide.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ Once you're familiar with the general idea, please start hacking...
 
 ## Quick start
 
+### Using Docker
+check [docker-guide](docs/docker-guide.md) and skip to the *Running mgmt* section
```
Comment (purpleidea) [inline docs/quick-start-guide.md]: Please use a full sentence with capitalization and period and move this section to the end after `Vagrant` server.

File: Makefile [hunk 3]
```diff
@@ -302,4 +302,16 @@ deb:
 	# cleanup
 	rm -rf debian/mgmt/
 
+build_container:
+	cd docker
+	docker build -t purpleidea/mgmgt-build -f ./Dockerfile.build .
+	docker run -td --name mgmt-build purpleidea/mgmgt-build
```
Comment (purpleidea) [inline Makefile]: still has the typo...

File: docs/quick-start-guide.md [hunk 4]
```diff
@@ -52,6 +52,10 @@ If you would like to avoid doing the above steps manually, we have prepared a
 project directory, run a `vagrant up`, and then a `vagrant status`. From there,
 you can `vagrant ssh` into the `mgmt` machine. The MOTD will explain the rest.
 
+### Using Docker
```
Comment (purpleidea) [inline docs/quick-start-guide.md]: Should be `## Docker` instead. Note the existing headings.

Comment (karmab) [reply docs/quick-start-guide.md]: done

File: Makefile (hunk 3 above)
Comment (karmab) [reply Makefile]: done

File: Makefile [hunk 5]
```diff
@@ -329,4 +329,16 @@ deb:
 	# cleanup
 	rm -rf debian/mgmt/
 
+build_container:
+	cd docker
+	docker build -t purpleidea/mgmt-build -f ./Dockerfile.build .
```
Comment (purpleidea) [inline Makefile]: isn't the path `docker/Dockerfile.build` ? Same below

**Comment by karmab** [comment]
@purpleidea 
- rebased on top of master
- fixed as per your requests
- tested successfully ( after adding additional dependencies in the building container) . the resulting latest binary can be found at docker.io/karmab/mgmt:06f2d65500b6e823b86fa45ec775a3b8f10c552c

**Comment by purpleidea** [comment]
Merged, thanks!

**Pull request merged**